Neos.Diff/Classes/Diff.php

```diff
@@ -0,0 +1,145 @@
+<?php
+namespace Neos\Diff;
+
+/**
+ * This file is part of the Neos.Diff package.
+ *
+ * (c) 2009 Chris Boulton <chris.boulton@interspire.com>
+ * Portions (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+/**
+ * Class Diff
+ */
+class Diff
+{
+    /**
+     * @var array The "old" sequence to use as the basis for the comparison.
+     */
+    private $a = null;
+
+    /**
+     * @var array The "new" sequence to generate the changes for.
+     */
+    private $b = null;
+
+    /**
+     * @var array Array containing the generated opcodes for the differences between the two items.
+     */
+    private $groupedCodes = null;
+
+    /**
+     * @var array Associative array of the default options available for the diff class and their default value.
+     */
+    private $defaultOptions = [
+        'context' => 3,
+        'ignoreNewLines' => false,
+        'ignoreWhitespace' => false,
+        'ignoreCase' => false
+    ];
+
+    /**
+     * @var array Array of the options that have been applied for generating the diff.
+     */
+    private $options = [];
+
+    /**
+     * The constructor.
+     *
+     * @param array $a Array containing the lines of the first string to compare.
+     * @param array $b Array containing the lines for the second string to compare.
+     * @param array $options Options (see $defaultOptions in this class)
+     */
+    public function __construct($a, $b, array $options = [])
```

Comment:
because the original library didn't have any. I only improved *some* of the code, and was rather happy that it would run on PHP 7. Well, that's how it is with external libraries.

But sure, let's improve it - now or later.